### PR TITLE
Remove conda environments after examples tests to prevent docker test from failing due to no-space-left error

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Remove conda environments created by the examples tests to prevent the docker test from faling due to a "No space left on device" error.

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,18 +5,18 @@ WORKDIR /app
 ADD . /app
 
 RUN apt-get update && \
-# install prequired modules to support install of mlflow and related components
+    # install prequired modules to support install of mlflow and related components
```
Comment (harupy) [inline Dockerfile:8]: Fix for triggering the docker test

File: .github/workflows/examples.yml [hunk 2]
```diff
@@ -40,6 +41,15 @@ jobs:
         source activate test-environment
         pytest --verbose tests/examples --large
 
+    - name: Clean up conda environments
+      run: |
+        conda env list |                        # list (env name, env path) pairs
+        cut -d' ' -f1 |                         # extract env names
+        grep "^mlflow-[a-z0-9]\{32\}" |         # filter envs created by mlflow
+        xargs -L 1 conda remove --all --yes -n  # pass env names to conda remove
```
Comment (harupy) [inline .github/workflows/examples.yml:49]: Maybe this command should be put in a shell script so we can use it  when we want to clean up all conda environments created by `mlflow run` for some reason.

Comment (aarondav) [reply .github/workflows/examples.yml:49]: Probably we should run something similar after the master tests workfow, as I think the large tests also create conda envs.

Comment (harupy) [reply .github/workflows/examples.yml:49]: Let me just chekc if the large tests really create conda envs.

Comment (harupy) [reply .github/workflows/examples.yml:49]: Looks like both small and large tests create some conda envs.

small: https://github.com/mlflow/mlflow/pull/3245/checks?check_run_id=967639307#step:5:9
large: https://github.com/mlflow/mlflow/pull/3245/checks?check_run_id=967639346#step:5:9

**Review by aarondav — APPROVED** [review]

File: .github/workflows/examples.yml (hunk 2 above)
Comment (harupy) [reply .github/workflows/examples.yml:49]: If the number of conda envs were large, we'd need to.

**Pull request merged**